Lib/unittest/test/testmock/testpatch.py

```diff
@@ -626,6 +626,13 @@ def test():
         self.assertEqual(foo.values, original)
 
 
+    def test_patch_dict_as_context_manager(self):
+        foo = {'a': 'b'}
+        with patch.dict(foo, a='c') as patched:
+            self.assertDictEqual(patched, {'a': 'c'})
```

Comment:
No need to use `assertDictEqual` explicitly here. `assertEqual` will use `assertDictEqual` if `a` and `b` both dicts:

https://github.com/python/cpython/blob/83c8675edb4fe278c5d930f7865977a5d3c7168a/Lib/unittest/case.py#L423